src/profile.c

```diff
@@ -239,6 +239,49 @@ static void CheckLeaveFunctionsAfterLongjmp(void)
     }
 }
 
+static Obj JsonEscapeString(Obj param)
+{
+    Int lenString = LEN_LIST(param);
+
+    // Massively over-long string
+    Obj     copy = NEW_STRING(lenString * 2 + 2);
+    UChar * in = CHARS_STRING(param);
+    UChar * base = CHARS_STRING(copy);
+    UChar * out = base;
+
+    for (Int i = 0; i <= lenString; ++i) {
+        UChar u = in[i];
+        switch (u) {
+        case '\\':
+        case '"':
+        case '/':
+            out[0] = '\\';
+            out[1] = u;
+            out += 2;
+            break;
+#define ESCAPE_CASE(x, y)                                                    \
+    case x:                                                                  \
+        out[0] = '\\';                                                       \
+        out[1] = y;                                                          \
+        out += 2;                                                            \
+        break;
+            ESCAPE_CASE('\b', 'b');
+            ESCAPE_CASE('\t', 't');
+            ESCAPE_CASE('\n', 'n');
+            ESCAPE_CASE('\f', 'f');
+            ESCAPE_CASE('\r', 'r');
+#undef ESCAPE_CASE
```

Comment:
Alternative solution, without macros:
```suggestion
        case '\b':
        case '\f':
        case '\n':
        case '\r':
        case '\t':
            out[0] = '\\';
            out[1] = u + ('a' - '\a');
            out += 2;
            break;
```

Reply:
wait, does this work? is this how they decided which character to call `\n`, `\t`, etc??

Reply:
No, I don't think it does, as `\n` is 10 and `\t` is 9 -- I don't think there is any way to map between `\n` and `n` without a hard-wired list somewhere (doesn't have to be in a macro of course)

Reply:
Argh, indeed, that was utter nonsense on my part, sorry. I got confused by the mapping of Ctrl-A, Ctrl-B, ... etc. to "ascii" codes 1, 2, ....